dataclients/kubernetes/kube.go

```diff
@@ -656,15 +668,27 @@ func (c *Client) convertPathRule(
 		eps, err = state.getEndpoints(ns, svcName, svcPort.String(), targetPort)
 		log.Debugf("convertPathRule: Found %d endpoints %s for %s", len(eps), targetPort, svcName)
 	}
+
+	if svc.Spec.Type == "ExternalName" {
```

Comment:
same question as above: wouldn't it be better to do this check before checking the target port and the endpoints?

Reply:
I see that the current approach relies on the target port.